romancal/resample/l3_wcs.py

```diff
@@ -39,44 +39,52 @@ def assign_l3_wcs(model, wcs):
     transform = wcs.forward_transform
 
     l3_wcsinfo.projection = "TAN"
-    l3_wcsinfo.image_shape = model.shape
+    l3_wcsinfo.image_shape = (
+        model.shape
+    )  # Should this be [::-1], pixel_shape is in cartesian coordinates
 
     # Fill out image-local information
     pixel_center = [(v - 1) / 2.0 for v in model.shape[::-1]]
     world_center = wcs(*pixel_center)
     l3_wcsinfo.ra = world_center[0]
     l3_wcsinfo.dec = world_center[1]
-    l3_wcsinfo.pixel_scale = compute_scale(wcs, world_center)
+    try:
+        l3_wcsinfo.pixel_scale_ref = transform.cdelt[0]
+    except AttributeError:
+        l3_wcsinfo.pixel_scale_ref = compute_scale(wcs, world_center)
     l3_wcsinfo.orientation = calc_pa(wcs, *world_center)
 
     footprint = create_footprint(wcs, model.shape, center=False)
     l3_wcsinfo.s_region = compute_s_region_keyword(footprint)
 
-    # Fill out wcs-general information
     try:
-        l3_wcsinfo.x_ref = -transform["crpix1"].offset.value
-        l3_wcsinfo.y_ref = -transform["crpix2"].offset.value
-    except IndexError:
+        l3_wcsinfo.x_ref = transform.crpix[0]
+        l3_wcsinfo.y_ref = transform.crpix[1]
+        world_ref = transform.crval
+    except AttributeError:
         log.warning(
             "WCS has no clear reference pixel defined by crpix1/crpix2. Assuming reference pixel is center."
         )
         l3_wcsinfo.x_ref = pixel_center[0]
         l3_wcsinfo.y_ref = pixel_center[1]
 
-    world_ref = wcs(l3_wcsinfo.x_ref, l3_wcsinfo.y_ref, with_bounding_box=False)
+        world_ref = wcs(l3_wcsinfo.x_ref, l3_wcsinfo.y_ref, with_bounding_box=False)
     l3_wcsinfo.ra_ref = world_ref[0]
     l3_wcsinfo.dec_ref = world_ref[1]
 
-    try:
-        cdelt1 = transform["cdelt1"].factor.value
-        cdelt2 = transform["cdelt2"].factor.value
-        l3_wcsinfo.pixel_scale_ref = (cdelt1 + cdelt2) / 2.0
-    except IndexError:
-        l3_wcsinfo.pixel_scale_ref = compute_scale(wcs, world_ref)
+    # pixel scale at center of image
+    l3_wcsinfo.pixel_scale = compute_scale(wcs, world_ref)
 
     l3_wcsinfo.orientation_ref = calc_pa(wcs, *world_ref)
 
-    l3_wcsinfo.rotation_matrix = transform["pc_rotation_matrix"].matrix.value.tolist()
+    try:
+        l3_wcsinfo.rotation_matrix = transform.pc.value.tolist()
+    except AttributeError:
+        log.warning(
+            "WCS has no clear rotation matrix defined by pc_rotation_matrix. Calculating one."
+        )
+        # rotation_matrix = utils.calc_rotation_matrix(l3_wcsinfo.orientat, 0.0)
+        # l3_wcsinfo.rotation_matrix = utils.list_1d_to_2d(rotation_matrix, 2)
```

Comment:
You don't actually calculate one in this case?  I agree that going through orientation seems right.  If we didn't make it required, not doing it also sounds okay.